ts/src/pro/okx.ts

```diff
@@ -392,7 +392,22 @@ export default class okx extends okxRest {
         // 2. Public depth channel, verification not required
         // 3. Data feeds will be delivered every 100ms (vs. every 200ms now)
         //
-        const depth = this.safeString (options, 'depth', 'books');
+        let depthChannel = 'books';
+        if (limit !== undefined) {
+            // limits are different depending endpoint: https://www.okx.com/docs-v5/en/#order-book-trading-market-data-ws-order-book-channel
+            const depthLimits = {
+                '1': 'bto-tbt',
+                '5': 'books5',
+                '50': 'books50-l2-tbt',
+                '400': 'books',
+            };
+            const limitString = this.numberToString (limit);
+            if (!(limitString in depthLimits)) {
+                throw new BadRequest (this.id + ' limit should be among: ' + this.json (depthLimits));
```

Comment:
@ttodua not sure about this, let's say the user passes 25 as the limit, couldn't we filter it after getting the ob?

Reply:
that's i also asked, I am not sure which way the lib should be. what is obvious, at this moment:
- we don't have any use of `limit` arg in okx
- either in none other exchanges we support custom limits, i.e. user inputed 23 ? none of existing exchanges support it and will throw error.

what should we decide with regard to that "global scale" problem?